File: API/Entities/AppUser.cs
using API.Extensions;

namespace API.Entities
{
    public class AppUser
    {
        public int Id { get; set; }
        public string UserName { get; set; }
        
        public byte[] PasswordHash { get; set; }
        public byte[] PasswordSalt { get; set; }
        public DateTime DateOfJoining { get; set; }
        public DateTime CreatedOn { get; set; } = DateTime.Now;
        public DateTime LastActive { get; set; } = DateTime.Now;
        public string Mobile { get; set; }
        public string Email { get; set; }
        public string Gender { get; set; }
        public string Designation { get; set; }
        public string City { get; set; }
        public string Country { get; set; }
        public ICollection<Photo> Photos { get; set; }

        // public int GetAge()
        // {
        //     return DateOfJoining.CalculateAge();
        // }
    }
}

File: Client/src/app/_models/appUser.ts
import { Photo } from "./photo";

export interface appUser {
        id: number;
        username: string;
        photoUrl: string;
        age: number;
        createdOn: Date;
        lastActive: Date;
        mobile: string;
        email: string;
        gender: string;
        designation: string;
        city: string;
        country: string;
        photos: Photo[];
    }